src/main/java/DukeException.java

```diff
@@ -0,0 +1,21 @@
+public abstract class DukeException extends Exception {
+    public DukeException(String s) {
```

Comment:
Perhaps you could use a more intuitive variable name, such as errorMessage, instead of s.